### ISSUE TITLE
[Feature Request] Sketcher - resize constraint icons independent from the font size

### ISSUE THREAD

**Issue by Intuos5** [FreeCAD/FreeCAD#8288]
### Is there an existing request for this?

- [X] I have searched the existing requests

### Forums discussion

https://forum.freecadweb.org/viewtopic.php?style=2&f=10&t=49972&start=70

### Subproject(s) affected?

Sketcher

### Idea description

I would like to resize the constraint icons independent from the font in the Sketcher workbench. The reason being, the icons are half the size of the font for me. I want to increase the constraint icons to improve readability, but I don't want very large dimension texts, because that deteriorates readability. 
Please add a dedicated constraint icon size parameter in the preferences of the Sketcher workbench. 

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by luzpaz** [comment]
You can attach images to the ticket. Before/After images are welcome.

**Comment by Intuos5** [comment]
Don't mind the silly sketch, it's for illustration purposes. Also note that this is on a 27 inch 4k screen.
Text size is correct, constraints are too small (text size is 20): 
![afbeelding](https://user-images.githubusercontent.com/63149655/215318049-00c1df51-20c9-4081-9314-acc42dd8d032.png)

Constraints are readable, but text is too large (text size is 40): 
![afbeelding](https://user-images.githubusercontent.com/63149655/215318113-730c30dd-ec87-42b3-bb6c-a9ff3871521a.png)

This is how I would like it to be (text size is 20, icon size is 40):
![preferable](https://user-images.githubusercontent.com/63149655/215318229-4f190f8c-7b67-4187-9f6b-bb8b5dacf2fc.png)

**Comment by luzpaz** [comment]
Not directly related but in the same vein to #7030

**Comment by LancashireDave** [comment]
I use a high res 43" monitor and the icons are so small I can hardly see them. The ability to change the icon size would be a great improvement

**Comment by luzpaz** [comment]
@Syres916 offered a possible solution in https://github.com/FreeCAD/FreeCAD/compare/master...Syres916:FreeCAD:patch-64

**Comment by luzpaz** [comment]
@PaddleStroke can you review proposed change ?

**Comment by PaddleStroke** [comment]
@Syres916 why keeping the dependency to the font size if we introduce a parameter for the size of icons ?

`    Client.drawingParameters.constraintIconSize = std::lround(sketchericonSizeRatio * 0.8 * sketcherfontSize);`

I think if we introduce a setting, we might as well remove the sketcherfontSize from the equation and just use a size preference for the icons.

**Comment by adrianinsaval** [comment]
I don't agree with adding a separate preference for this, instead it looks to me like there's a problem with how the text and the symbols are scaling at different rates for this person and it should be fixed to be more sensible. Here's how it scales for me, at 40px:
![image](https://github.com/FreeCAD/FreeCAD/assets/36372335/806df8b6-e812-4d2c-a266-57755a9d1cbd)
at 20 px:
![image](https://github.com/FreeCAD/FreeCAD/assets/36372335/e67b764f-850c-40cd-84cd-f6ef7c34a298)

**Comment by Intuos5** [comment]
If possible, I would prefer independent scaling of text and icons, because I prefer to set them at scales that make it most comfortable for me. I may want smaller text and larger icons for legibility, others may want the same size, etc.

**Comment by adrianinsaval** [comment]
if implemented I would suggest it be through a parameter only, let's not keep polluting preference pages with obscure options to satisfy individual wishes.

**Comment by luzpaz** [comment]
Is this on the 1.0 roadmap ?

**Comment by ChrisAbra** [comment]
> keep polluting preference pages with obscure options to satisfy individual wishes.

Seems strange that this obvious accessibilty feature would be the line to draw.... 

It also makes sense to tie it somewhat to font-size as the default behaviour would want them scaling together but the ability to override is important too as clearly demonstrated above.

**Comment by adrianinsaval** [comment]
> Seems strange that this obvious accessibility feature would be the line to draw....

the ability to scale is an accessibility feature, this already exists. In what situation do you need to scale one but not the other?

**Comment by ChrisAbra** [comment]
I think its reasonable to want to have legibly sized constraints icons without having giant text as demonstrated in the images above. There are multiple users who say this will benefit them and one saying they dont like it for "clutter" or "works on my machine" reasons.

"Youre doing it wrong" cant keep being the FreeCAD approach

**Comment by adrianinsaval** [comment]
see my previous comment: https://github.com/FreeCAD/FreeCAD/issues/8288#issuecomment-1753607495
